airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -390,5 +420,35 @@ def _iter_upstream_conditions() -> Iterator[ColumnOperators]:
                         f"upstream_task_ids={task.upstream_task_ids}"
                     )
                 )
+        elif trigger_rule == TR.ALL_DONE_SETUP_SUCCESS:
+            status = None
+            if not upstream_done:
+                status = self._failing_status(
```

Comment:
```suggestion
                yield self._failing_status(
```

Should these just be yielding from here? Does the extra variable buy us anything?

Reply:
I refactored this.